### PR TITLE
Fix: break out of recursive loops on first focus

### PR DESCRIPTION
**Pull request by DPflasterer**
I have a form structure like so, where I have a group depth of more than 2.

```
personal.name.first
personal.name.last
personal.phone-number

```
What was happening is when `iterateFieldsByAction` calls itself recursively it did call focus on `personal.name.first` but then it breaks out of `personal.name` and continues looping and calling focus on `personal.phone-number`.

This change will break out of the current loop and recursively break out of all loops.

### CODE REVIEW

File: src/__tests__/logic/iterateFieldsByAction.test.ts [hunk 1]
```diff
@@ -64,4 +64,145 @@ describe('focusFieldBy', () => {
...
+      },
+      (ref) => {
+        // @ts-ignore
+        ref.focus && ref.focus(ref.name);
+        return 1;
+      },
+    );
+    expect(focus).toBeCalledWith('first');
+    expect(focus).not.toBeCalledWith('second');
+  });
```
Comment (bluebill1049) [inline src/__tests__/logic/iterateFieldsByAction.test.ts:99]: existing logic pass this test as well.

File: src/__tests__/logic/iterateFieldsByAction.test.ts [hunk 2]
```diff
@@ -64,4 +64,145 @@ describe('focusFieldBy', () => {
...
+          // @ts-ignore
+          ref.focus && ref.focus(ref.name);
+          return 1;
+        }
+        return;
+      },
+    );
+    expect(focus).not.toBeCalledWith('first');
+    expect(focus).toBeCalledWith('last');
+  });
```
Comment (bluebill1049) [inline src/__tests__/logic/iterateFieldsByAction.test.ts:140]: existing logic pass this test as well.

**Review by bluebill1049 — COMMENTED** [review]
not sure the fix here, i thought the existing logic to break early as well

File: src/__tests__/logic/iterateFieldsByAction.test.ts (hunk 1 above)
Comment (DPflasterer) [reply src/__tests__/logic/iterateFieldsByAction.test.ts:99]: That is correct. These first few tests were simply to improve code coverage by asserting focus isn't called on when it's not expected inputs. The`should should recursively drill into objects and break out of all loops on first focus` test fixes the issue solved by this PR.

File: src/__tests__/logic/iterateFieldsByAction.test.ts (hunk 2 above)
Comment (DPflasterer) [reply src/__tests__/logic/iterateFieldsByAction.test.ts:140]: That is correct. These first few tests were simply to improve code coverage by asserting focus isn't called on when it's not expected inputs.

**Comment by DPflasterer** [comment]
> not sure the fix here, i thought the existing logic to break early as well

The existing logic does break early out of the first loop but it is not breaking out of the recursion. The last test case added, fixes the issue which I'll  try to illustrate.

Before:
```
personal
    name
        first
        last <- called here and breaks out of 'name'
    phone <- called here and breaks out of 'personal'
    address
        line1
```

After:
```
personal
    name
        first
        last <- called here and returns false to 'personal' and breaks out of it
    phone
    address
        line1
```

**Pull request merged**